lib/vault/encrypted_model.rb

```diff
@@ -247,12 +232,16 @@ def __vault_persist_attribute!(attribute, options)
 
         # Only persist changed attributes to minimize requests - this helps
         # minimize the number of requests to Vault.
-        if !changed.include?("#{attribute}")
-          return
+        if ActiveRecord::VERSION::STRING.to_f >= 5.2
+          return unless previous_changes_include?(attribute)
+        elsif ActiveRecord::VERSION::STRING.to_f >= 5.1
+          return unless saved_change_to_attribute?(attribute.to_s)
+        else
+          return unless attribute_changed?(attribute)
```

Comment:
Rather than coerce the version string to a float, it would be safer to use `ActiveRecord.gem_version` and create a new `Gem::Version` to compare them to:

```rb
ActiveRecord.gem_version >= Gem::Version.new("5.2")
```